### PR TITLE
Create Seperate Audio Class

### PR DESCRIPTION
**Pull request by TrentGoodall**
[WIP] Something I am working on that targets all audio related functions, files and sound sets. All current functions & overloads are working but I am still testing audio files and sound sets to expand this class even further.

### CODE REVIEW

**Comment by IncoCode** [comment]
Why not add Audio.cpp and Audio.hpp in one pull request? :confused:

**Comment by TrentGoodall** [comment]
Apologies @IncoCode (and everyone else). This is my first time using GH so I accidentally created 2 pull requests.

**Comment by crosire** [comment]
No problem at all. Just merge your patch-1 branch into patch-2 and this pull request is automatically updated with Audio.hpp =)

**Comment by crosire** [comment]
What are the two enums for? There aren't used anywhere as far as I can see.

**Comment by TrentGoodall** [comment]
The enums are mostly there for testing purposes at the moment, but will be added as parameters for several of the audio methods later on. 

I have created an audio queue script (in my dev branch) and am currently testing all sound sets and sound files in order to expand upon the Audio class. I am also hoping to access the police scanner audio files (which I am yet to decrypt) to create a police scanner sub class.

**Comment by JohnnyCrazy** [comment]
Any problem with this one @crosire ? Seems like it's all fine :)

**Comment by crosire** [comment]
I would move the functions into "World".

File: source/Audio.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#include "Audio.hpp"
+#include "Native.hpp"
+#include "Entity.hpp"
+
+namespace GTA
+{
+	int Audio::GetSoundID()
+	{
+		return Native::Function::Call<int>(Native::Hash::GET_SOUND_ID);
```
Comment (JohnnyCrazy) [inline source/Audio.cpp]: Any info on how to use this? How do we get the ID of sounds which other methods use? (like HasSoundFinished)

**Comment by JohnnyCrazy** [comment]
Mh...I wonder if the World class won't get too bloated with both enums.

**Comment by crosire** [comment]
Those enums aren't used right now, so I'd just remove them. All those enums are bloating the code anyway, strings are easier to handle (even though they are slower).

File: source/Audio.cpp (hunk 1 above)
Comment (TrentGoodall) [inline source/Audio.cpp]:
```c#
int id = Audio.GetSoundID();
Audio.PlaySoundFrontEnd(id, "PIN_BUTTON", "ATM_SOUNDS");
// In a script tick loop
if (Audio.HasSoundFinished(id)) {
    Audio.ReleaseSoundId(id);
    // Handle sound finished...
}
```

**Pull request closed without merging**